saleor/graphql/checkout/mutations.py

```diff
@@ -85,9 +85,14 @@ def update_checkout_shipping_method_if_invalid(checkout: models.Checkout, discou
         checkout.shipping_method = None
         checkout.save(update_fields=["shipping_method"])
 
-    is_valid = clean_shipping_method(
-        checkout=checkout, method=checkout.shipping_method, discounts=discounts
-    )
+    is_valid = True
+    try:
+        is_valid = clean_shipping_method(
+            checkout=checkout, method=checkout.shipping_method, discounts=discounts
+        )
+    except ValidationError:
```

Comment:
Not 100% sure about this - all `ValidationError`s we raise in Graphene are supposed to be returned as errors in API. If we're catching it here, it means that maybe we shouldn't raise it in `clean_shipping_method`?  Maybe resetting the shipping method should happen in `update_checkout_shipping_method_if_invalid`?

Reply:
How about the current solution?